lua/nvim_lsp/configs.lua

```diff
@@ -33,26 +33,26 @@ function configs.__newindex(t, config_name, config_def)
   -- The config here is the one which will be instantiated for the new server,
   -- which is why this is a function, so that it can refer to the settings
   -- object on the server.
-  local function add_callbacks(config)
-    config.callbacks["window/logMessage"] = function(err, method, params, client_id)
+  local function add_handlers(config)
+    assert(not config.callbacks, "lsp.callbacks has been deprecated. See here for more: https://github.com/neovim/neovim/pull/12655")
+    config.handlers["window/logMessage"] = function(err, method, params, client_id)
       if params and params.type <= config.log_level then
         -- TODO(ashkan) remove this after things have settled.
-        assert(lsp.callbacks, "Update to Nvim HEAD. This is an incompatible interface.")
-        assert(lsp.callbacks["window/logMessage"], "Callback for window/logMessage notification is not defined")
-        lsp.callbacks["window/logMessage"](err, method, params, client_id)
+        assert(lsp.handlers["window/logMessage"], "Handler for window/logMessage notification is not defined")
+        lsp.handlers["window/logMessage"](err, method, params, client_id)
       end
     end
 
-    config.callbacks["window/showMessage"] = function(err, method, params, client_id)
+    config.handlers["window/showMessage"] = function(err, method, params, client_id)
       if params and params.type <= config.message_level then
         -- TODO(ashkan) remove this after things have settled.
-        assert(lsp.callbacks and lsp.callbacks[method], "Update to Nvim HEAD. This is an incompatible interface.")
-        assert(lsp.callbacks["window/showMessage"], "Callback for window/showMessage notification is not defined")
-        lsp.callbacks["window/showMessage"](err, method, params, client_id)
+        assert(lsp.handlers and lsp.handlers[method], "Update to Nvim HEAD. This is an incompatible interface.")
```

Comment:
I don't think is needed anymore.

Reply:
You're right. If we're aiming to assert on addition, we don't need to ship back a function which makes assertions about that again.